gwtp-core/gwtp-mvp-client/src/main/java/com/gwtplatform/mvp/client/ViewImpl.java

```diff
@@ -32,17 +44,62 @@
  */
 public abstract class ViewImpl implements View {
     private Widget widget;
+    private Map<Object, HasOneWidget> oneWidgetSlots = new HashMap<Object, HasOneWidget>();
+
+    private Map<Object,HasWidgets> hasWidgetSlots = new HashMap<Object, HasWidgets>();
 
+    private Map<OrderedSlot<?>, List<Comparable<Comparable<?>>>> orderedSlots
+            = new HashMap<OrderedSlot<?>, List<Comparable<Comparable<?>>>>();
+
+    @SuppressWarnings("unchecked")
     @Override
     public void addToSlot(Object slot, IsWidget content) {
+        if (hasWidgetSlots.containsKey(slot)) {
+            if (orderedSlots.containsKey(slot)) {
```

Comment:
This was actually correct, so I'll need to revert to it.

hasWidgetSlots and orderedSlots store different things.

hasWidgetSlots stores a map of the slot to the panel.

orderedSlots stores a map to a list of the presenters that are in the slot.

The orderedSlot list is able to be sorted and then from there it can work out where in the panel the widget should be inserted.